detectron2/evaluation/coco_evaluation.py

```diff
@@ -49,6 +52,15 @@ def __init__(self, dataset_name, cfg, distributed, output_dir=None):
         self._logger = logging.getLogger(__name__)
 
         self._metadata = MetadataCatalog.get(dataset_name)
+
+        PathManager.mkdirs(output_dir)
+        if not hasattr(self._metadata, "json_file"):
+            self._logger.warning(f"json_file was not found in MetaDataCatalog for [{dataset_name}]")
+
+            cache_path = convert_to_coco_json(dataset_name, output_dir)
+            self._metadata.json_file = cache_path
+
+        # TODO: json_file should be renamed to json_path
```

Comment:
I don't get the TODO here: I think using "xx_file" for file path is quite common and renaming something users are depending on is not easy.

Reply:
I think that file variable names are actually reserved for file objects and not strings, but these are just nuances.

Reply:
removed TODO in 021ec57 as requested